src/librustc/infer/error_reporting/mod.rs

```diff
@@ -994,6 +996,21 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                         self.push_comma(&mut values.0, &mut values.1, len, i);
                     }
 
+                    // Do the same for const arguments, if they are equal, do not highlight and
+                    // elide them from the output.
+                    let const_arguments = sub1.consts().zip(sub2.consts());
+                    for (i, (ca1, ca2)) in const_arguments.enumerate() {
+                        let i = i + consts_offset;
+                        if ca1 == ca2 {
+                            values.0.push_normal("_");
```

Comment:
are there tests for this path? where some other error happens when you have `T<"a", "b"> != T<"A","b">` for example (or even for when you have const generics that are equal but other things wrong?

Reply:
I have added tests for this.